lib/FrontendTool/FrontendTool.cpp

```diff
@@ -761,10 +762,16 @@ static bool performCompile(std::unique_ptr<CompilerInstance> &Instance,
   }
 
   if (opts.ValidateTBDAgainstIR) {
-    bool validationError =
-        PrimarySourceFile ? validateTBD(PrimarySourceFile, *IRModule)
-                          : validateTBD(Instance->getMainModule(), *IRModule);
-    if (validationError)
+    auto hasMultipleIRGenThreads = Invocation.getSILOptions().NumThreads > 1;
```

Comment:
I'm just passing through all the options that IRGen is asking for ( https://github.com/apple/swift/blob/d1c5de1cea6c388c78266cc1b691074d06b6eb33/lib/IRGen/GenDecl.cpp#L1372-L1386 ), to make sure we're matching as closely as possible. If you think this isn't required I'd be very happy to get rid of it.